### PR TITLE
Fix `tootctl accounts cull`

### PR DESCRIPTION
**Pull request by BenLubar**
Prevent unreachable domains from inheriting the previous status code.

Also list the actual accounts that would have been culled during a dry run.
(Otherwise, the dry run mode is basically useless.)

### CODE REVIEW

**Comment by BenLubar** [comment]
(sorry Code Climate robot, I'm too used to bash scripting)

**Comment by ashleyhull-versent** [comment]
doesn't this job take multiple hours to complete?

File: lib/mastodon/accounts_cli.rb [hunk 1]
```diff
@@ -219,12 +219,14 @@ def backup(username)
...
       skip_threshold = 7.days.ago
       culled         = 0
+      dry_run_culled = []
       skip_domains   = Set.new
       dry_run        = options[:dry_run] ? ' (DRY RUN)' : ''
 
       Account.remote.where(protocol: :activitypub).partitioned.find_each do |account|
         next if account.updated_at >= skip_threshold || (account.last_webfingered_at.present? && account.last_webfingered_at >= skip_threshold)
 
+        code = 0
```
Comment (ThibG) [inline lib/mastodon/accounts_cli.rb]: I would have put a `next` like in the other `rescue` instead of that, but LGTM

**Review by ClearlyClaire — APPROVED** [review]

**Comment by Gargron** [comment]
Please modify the changelog with the corresponding message as well

File: CHANGELOG.md [hunk 2]
```diff
@@ -75,6 +76,7 @@ All notable changes to this project will be documented in this file.
 - Fix race condition when streaming out deleted statuses ([ThibG](https://github.com/tootsuite/mastodon/pull/10280))
 - Fix performance of admin federation UI by caching account counts ([Gargron](https://github.com/tootsuite/mastodon/pull/10374))
 - Fix JS error on pages that don't define a CSRF token ([hinaloe](https://github.com/tootsuite/mastodon/pull/10383))
+- Fix unreachable domains sometimes deleting accounts in `tootctl accounts cull` ([BenLubar](https://github.com/tootsuite/mastodon/pull/10460))
```
Comment (Gargron) [inline CHANGELOG.md]: Fix `tootctl accounts cull` sometimes removing accounts that are temporarily unreachable

File: CHANGELOG.md [hunk 3]
```diff
@@ -50,6 +50,7 @@ All notable changes to this project will be documented in this file.
 - Change Webpack to not use @babel/preset-env to compile node_modules ([ykzts](https://github.com/tootsuite/mastodon/pull/10289))
 - Change web UI to use new Web Share Target API ([gol-cha](https://github.com/tootsuite/mastodon/pull/9963))
 - Change ActivityPub reports to have persistent URIs ([ThibG](https://github.com/tootsuite/mastodon/pull/10303))
+- Changed `tootctl accounts cull --dry-run` to show a list of accounts that would be deleted ([BenLubar](https://github.com/tootsuite/mastodon/pull/10460))
```
Comment (Gargron) [inline CHANGELOG.md]: Change `tootctl accounts cull --dry-run` to list of accounts that would be deleted

(Change, not changed)

**Review by Gargron — APPROVED** [review]

**Pull request merged**